### PR TITLE
generate proposal labels doc

### PR DESCRIPTION
**Pull request by xingyuanbu**
complete generate proposal labels document
#12939

### CODE REVIEW

File: paddle/fluid/operators/detection/generate_proposal_labels_op.cc [hunk 1]
```diff
@@ -439,31 +439,88 @@ class GenerateProposalLabelsKernel : public framework::OpKernel<T> {
...
+    AddAttr<bool>(
+        "use_random",
+        "Use random sampling to choose foreground and background boxes.")
+        .SetDefault(true);
 
     AddComment(R"DOC(
-Generate Proposals Labels Operator.
-)DOC");
+This operator can be, for given the GenerateProposalOp output bounding boxes and groundtruth,
+to sample foregroud boxes and background boxes, and compute loss target.
```
Comment (jerrywgz) [inline paddle/fluid/operators/detection/generate_proposal_labels_op.cc]: foreground

File: paddle/fluid/operators/detection/generate_proposal_labels_op.cc [hunk 2]
```diff
@@ -439,31 +439,88 @@ class GenerateProposalLabelsKernel : public framework::OpKernel<T> {
...
+
+RpnRois is the output boxes of RPN and was processed by generate_proposal_op, these boxes
+were combined with groundtruth boxes and sampled according to batch_size_per_im and fg_fraction,
+If an instance with a groundtruth overlap greater than fg_thresh, then it was considered as a foregroud sample.
+If an instance with a groundtruth overlap greater than bg_thresh_lo and lower than bg_thresh_hi,
+then it was considered as a background sample.
+After all foregroud and background boxes are chosen (so called Rois),
```
Comment (jerrywgz) [inline paddle/fluid/operators/detection/generate_proposal_labels_op.cc]: foreground

File: python/paddle/fluid/layers/detection.py [hunk 3]
```diff
@@ -1413,7 +1413,36 @@ def generate_proposal_labels(rpn_rois,
...
+    to sample foregroud boxes and background boxes, and compute loss target.
+
+    RpnRois is the output boxes of RPN and was processed by generate_proposal_op, these boxes
+    were combined with groundtruth boxes and sampled according to batch_size_per_im and fg_fraction,
+    If an instance with a groundtruth overlap greater than fg_thresh, then it was considered as a foregroud sample.
+    If an instance with a groundtruth overlap greater than bg_thresh_lo and lower than bg_thresh_hi,
+    then it was considered as a background sample.
+    After all foregroud and background boxes are chosen (so called Rois),
+    then we apply random sampling to make sure
+    the number of foregroud boxes is no more than batch_size_per_im * fg_fraction.
```
Comment (jerrywgz) [inline python/paddle/fluid/layers/detection.py]: foreground

**Review by jerrywgz — DISMISSED** [review]
LGTM

**Review by qingqing01 — DISMISSED** [review]

**Review by qingqing01 — APPROVED** [review]

**Pull request merged**